securedrop/source_templates/lookup.html

```diff
@@ -1,31 +1,34 @@
 {% extends "base.html" %}
-{% block body %}
 
+{% block body %}
 {% if new_user %}
-<div class="code-reminder pull-left" id="codename-hint">
+<section class="code-reminder pull-left" id="codename-hint" aria-label="{{ gettext('Codename') }}">
   <div id="codename-hint-visible">
-    <img class="pull-left" src="{{ url_for('static', filename='i/font-awesome/lock-black.png') }}" alt="" width="17" height="20"> {{ gettext('Remember, your codename is:') }}
-    <a id="codename-hint-show" class="show pull-right visible-codename" href="#codename-hint-visible">{{ gettext('Show') }}</a>
+    <label id="codename-label" for="codename" aria-hidden="true">{{ gettext('Remember, your codename is:') }}</label>
+    {# ARIA-HIDDEN violates axe rule aria-hidden-focus, because we (a) want to
+    hide the superfluous "show"/"hide" links from screen-readers, (b) do not
+    want to remove these elements from sequential navigation with TABINDEX="-1",
+    and (c) do not have recourse in the Source Interface to scripting the
+    ARIA-HIDDEN value dynamically.  Cf. #6031. #}
+    <a id="codename-hint-show" class="show pull-right visible-codename" href="#codename-hint-visible" aria-hidden="true" tabindex="-1">{{ gettext('Show') }}</a>
     <div id="codename-hint-content" class="hidden-codename codename">
-      <a id="codename-hint-hide" class="pull-right" href="#codename-hint">{{ gettext('Hide') }}</a>
-      <p>{{ codename }}</p>
+      <a id="codename-hint-hide" class="pull-right" href="#codename-hint" aria-hidden="true">{{ gettext('Hide') }}</a>
```

Comment:
I had looked into a `details`/`summary` implementation of this widget and ruled it out—and now I can't reconstruct why.  I have a feeling I was overdoing the `aria-*` attributes in a way that interfered with my screen-reader's parsing of the native semantics and behavior.

So, I've added this properly, I think, in 66ab8ba1f.  As I note in the commit, I would have preferred to override the browser's native styling of the `::marker` pseudo-element, but its placement relative to the `summary` content depends on the language direction.  So this implementation uses `::{before,after}` as you suggest.  The downside is that even with `@media only screen` some screen-readers (including mine) will announce the `::{before,after}` `content` values (the `Show` and `Hide` labels) in addition to the automatic cues about the state of the `details` element.

Reply:
Yes, this semantic is a little problematic. The ideal solution would be something like:

```html
<p>Remember, you codename is:</p>
<details>
    <summary>Toggle visibility</summary>
    <output>{{ codename }} </output>
</details>
```
So, in that case no CSS hack would be needed. Or with CSS hack, instead of "Toggle visibility" I would have "Show" and then `[open] summary` I can hide, and `[open] summary:after` can have "Hide". And then also control the output's visibility to assistive tech via the `[open]` state. My reasoning kind of behind that was the codename shouldn't be announced all the time and the user using a screen reader should also be able to toggle between the functionality.

But I think this might be an entire separate discussion of UX and what the team thinks is ideal, so as long as this doesn't cause any regression in the screen-reader announcements, I think it's still much cleaner than the anchor hack. @cfm what do you think?